packages/form-layout/src/vaadin-form-layout-mixin.js

```diff
@@ -412,29 +421,67 @@ export const FormLayoutMixin = (superClass) =>
 
     /** @private */
     __updateCSSGridLayout() {
+      let columnCount = 0;
+      let maxColumns = 0;
+
+      this.$.layout.style.setProperty('--_rendered-column-count', this.__renderedColumnCount);
       const fitsLabelsAside = this.offsetWidth >= this.__columnWidthWithLabelsAside;
       this.$.layout.toggleAttribute('fits-labels-aside', this.labelsAside && fitsLabelsAside);
 
-      let resetColumn = false;
-      [...this.children]
-        .flatMap((child) => {
-          return child.localName === 'vaadin-form-row' ? [...child.children] : child;
-        })
-        .forEach((child) => {
-          const { parentElement } = child;
+      this.__children
+        .filter((child) => isBreakLine(child) || !isElementHidden(child))
+        .forEach((child, index, children) => {
+          const prevChild = children[index - 1];
 
-          if (child.localName === 'br' || parentElement.localName === 'vaadin-form-row') {
-            resetColumn = true;
+          if (isBreakLine(child)) {
+            columnCount = 0;
             return;
           }
 
-          if (resetColumn && !isElementHidden(child) && parentElement.localName !== 'vaadin-form-row') {
+          if (
+            (prevChild && prevChild.parentElement !== child.parentElement) ||
+            (!this.autoRows && child.parentElement === this)
+          ) {
+            columnCount = 0;
+          }
+
+          if (columnCount === 0) {
```

Comment:
I wonder if this CSS property doesn't actually need to be set when autoRows is off

```suggestion
          if (this.autoRows && columnCount === 0) {
```

Reply:
Makes sense. Updated the snapshots as well since this change affects them.